### PR TITLE
fix: authentication check for the chatbot and review pages

### PR DESCRIPTION
**Pull request by GolimarGit**
### 🛠 Proposed Changes:

The `NavBar` navigation links use a mouse event instead of directly changing the `href`.
Session token checks added to the chatbot and review routes.

### 📝 Details:

- `"src/lib/components/NavBar.svelte"`: the `TabAnchor` tag uses the `on:click={...}` mouse event rather than `href={href.link}`
- `"src/routes/api/chatbot/+server.ts"`: session retrieved and verified in POST method
- `"src/routes/review/+page.svelte"`: page redirect check added
- `"src/routes/api/review/+server.ts"`: session retrieved and verified in POST method

### 🔗 Related Issue(s):

N/A

### CODE REVIEW

**Comment by Verdone** [comment]
Try running `npm run lint` and `npm run format`, then committing the necessary changes.

**Comment by GolimarGit** [comment]
the code should be _prettier_ now :pepe_laugh:

File: prisma/schema/Review.prisma
Comment (briantkatch) [inline prisma/schema/Review.prisma:1]: If you're changing the schema, you need to include a Prisma migration.

File: src/routes/api/review/+server.ts
Comment (briantkatch) [inline src/routes/api/review/+server.ts:1]: You don't need to validate the session yourself to check auth (and the cookie isn't called "session"). Use the local provided by hooks.server.ts.

**Review by briantkatch — CHANGES_REQUESTED** [review]
- Must include Prisma migration
- Do not manually decode the cookie; use the local provided by hooks.server.ts

**Comment by briantkatch** [comment]
See the server code in routes/admin/page.server.ts for an example of imposing an auth requirement.

**Comment by Niravanaa** [comment]
Authentication checks do not need to be done as the pages are already auth-protected through `hooks.server.ts`.

The schema is also being changed, but that is not what the intended scope of the PR should be.

The only change that needs to be done is the TabAnchor's on:click event being changed, nothing else really.

I'd recommend opening a new PR with that sole fix, instead of reverting any of the changes done in this PR

@GolimarGit

**Comment by GolimarGit** [comment]
I did not intend to modify the schema; it was probably changed from the `npx prettier --write` ([commit details](https://github.com/SvelteShipSolutions/SvelteShip-SOEN-343/pull/34/commits/25a54258a02f1e79bac91defb358ff3f131537ef)).
If authentication checks do not need to be done directly in the separate pages, I will make a new branch that just changes the `on:click` event in the `TabAnchor` in `src/lib/components/NavBar.svelte`.
My fault gang.

**Pull request closed without merging**